lua/gitlab/actions/discussions.lua

```diff
@@ -14,32 +14,401 @@ local miscellaneous = require("gitlab.actions.miscellaneous")
+      if first_note.position.line_range ~= nil then
+        -- Diagnostics for line range discussions are tricky - you need to set lnum to
+        -- line number equal to note.position.new_line or note.position.old_line because that is
+        -- only line where you can trigger the diagnostic show. This also need to be in sinc
+        -- with the sign placement.
+        local start_old_line, start_new_line = _parse_line_code(first_note.position.line_range.start.line_code)
+        local end_old_line, end_new_line = _parse_line_code(first_note.position.line_range["end"].line_code)
+        if first_note.position.line_range.start.type == "new" then
+          local new_diagnostic
+          if first_note.position.new_line == start_new_line then
+            new_diagnostic = {
+              lnum = start_new_line - 1,
+              end_lnum = end_new_line - 1,
+            }
+          else
+            new_diagnostic = {
+              lnum = end_new_line - 1,
+              end_lnum = start_new_line - 1,
+            }
+          end
+          new_diagnostic = vim.tbl_deep_extend("force", new_diagnostic, diagnostic)
+          table.insert(new_diagnostics, new_diagnostic)
+        elseif first_note.position.line_range.start.type == "old" then
+          local old_diagnostic
+          if first_note.position.old_line == start_old_line then
+            old_diagnostic = {
+              lnum = start_old_line - 1,
+              end_lnum = end_old_line - 1,
+            }
+          else
+            old_diagnostic = {
+              lnum = end_old_line - 1,
+              end_lnum = start_old_line - 1,
+            }
+          end
+          old_diagnostic = vim.tbl_deep_extend("force", old_diagnostic, diagnostic)
+          table.insert(old_diagnostics, old_diagnostic)
+        end
+      else
+        -- Diagnostics for single line discussions.
+        if first_note.position.new_line ~= nil then
+          local new_diagnostic = {
+            lnum = first_note.position.new_line - 1,
+          }
+          new_diagnostic = vim.tbl_deep_extend("force", new_diagnostic, diagnostic)
+          table.insert(new_diagnostics, new_diagnostic)
+        end
+        if first_note.position.old_line ~= nil then
+          local old_diagnostic = {
+            lnum = first_note.position.old_line - 1,
+          }
+          old_diagnostic = vim.tbl_deep_extend("force", old_diagnostic, diagnostic)
+          table.insert(old_diagnostics, old_diagnostic)
+        end
+      end
+    end
+  end
+
+  vim.diagnostic.reset(diagnostics_namespace)
+  reviewer.set_diagnostics(
+    diagnostics_namespace,
+    new_diagnostics,
+    "new",
+    state.settings.discussion_diagnostics.display_opts
+  )
+  reviewer.set_diagnostics(
+    diagnostics_namespace,
+    old_diagnostics,
+    "old",
+    state.settings.discussion_diagnostics.display_opts
+  )
+end
+
+---Refresh discussion data, discussion signs and diagnostics
+M.refresh_discussion_data = function()
```

Comment:
Can we please add an invocation of this function in the `lua/gitlab/actions/comment.lua` file, after both notes and comments are created? This will make the comment appear in the buffer right away.